packages/mobile/src/screens/favorites-screen/TracksTab.tsx

```diff
@@ -146,47 +224,60 @@ export const TracksTab = () => {
     [playingUid, isPlaying, dispatch, isReachable, isOfflineModeEnabled]
   )
 
-  const isLoading = savedTracksStatus !== Status.SUCCESS
-  const tracks = savedTracks.entries as TrackMetadata[]
-  const hasNoFavorites = tracks.length === 0
+  const filteredTracks = allTracksFetched
+    ? tracks.filter(filterTrack)
+    : tracks.filter((track) => track.kind !== Kind.EMPTY)
 
   return (
-    <WithLoader loading={isLoading}>
-      <VirtualizedScrollView listKey='favorites-screen'>
-        {!isLoading && hasNoFavorites && !filterValue ? (
-          isOfflineModeEnabled && !isReachable ? (
-            <NoTracksPlaceholder />
-          ) : (
-            <EmptyTileCTA message={messages.emptyTabText} />
-          )
+    <VirtualizedScrollView listKey='favorites-screen'>
+      {!isLoading && tracks.length === 0 && !filterValue ? (
+        isOfflineModeEnabled && !isReachable ? (
+          <NoTracksPlaceholder />
         ) : (
-          <>
-            <OfflineContentBanner />
-            <FilterInput
-              value={filterValue}
-              placeholder={messages.inputPlaceholder}
-              onChangeText={setFilterValue}
-            />
-            {tracks.length ? (
-              <Tile
-                styles={{
-                  root: styles.container,
-                  tile: styles.trackListContainer
-                }}
-              >
-                <TrackList
-                  onSave={onToggleSave}
-                  showDivider
-                  togglePlay={togglePlay}
-                  trackItemAction='save'
-                  tracks={tracks.filter(filterTrack)}
-                  hideArt
+          <EmptyTileCTA message={messages.emptyTabText} />
+        )
+      ) : (
+        <>
+          <OfflineContentBanner />
+          <FilterInput
+            value={filterValue}
+            placeholder={messages.inputPlaceholder}
+            onChangeText={setFilterValue}
+          />
+          <WithLoader loading={initialFetch}>
```

Comment:
Personally would like to see skeletons render initially, would that be hard?

Reply:
More of a product question I suppose. Probably not too difficult, but I was trying to not change things too much

Reply:
Yeah for this go around no issues, just wanted to air me thoughts in case you felt similar

Reply:
Yea I like the idea. I'll add an issue in linear to update this later